src/utils.js

```diff
@@ -71,36 +71,71 @@ const _runJavascript = (jsString, window, filename = 'script', lineOffset = 0, c
 };
 module.exports._runJavascript = _runJavascript;
 
-const _normalizeBuffer = (b, target) => {
-  const name = b && b.constructor && b.constructor.name;
-  switch (name) {
-    case 'Buffer':
-    case 'ArrayBuffer':
-    case 'Uint8Array':
-    case 'Uint8ClampedArray':
-    case 'Int8Array':
-    case 'Uint16Array':
-    case 'Int16Array':
-    case 'Uint32Array':
-    case 'Int32Array':
-    case 'Float32Array':
-    case 'Float64Array':
-    case 'DataView':
-    case 'Promise': {
-      if (!(b instanceof target[name])) {
-        GlobalContext.nativeVm.setPrototype(b, target[name].prototype);
-      }
-      break;
+/**
+ * Convert (normalize prototype) native object to pass to JS.
+ * Or convert JS object when passing to native.
+ * Node's vm module was slow so there's vm-one for one-way bindings.
+ * github.com/modulesio/vm-one
+ * nativeVm.setPrototype will do the nan casting setPrototype between types.
+ * Required for instanceof to work.
+ *
+ * @param obj - Object to convert.
+ * @param targetContext - window or vm context that prototype will get retrieved from.
+ */
+const _normalizeBuffer = (function () {
```

Comment:
`_normalizeBuffer` is probably a misnomer. It's more like `_convertNativePrototype`.